core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java

```diff
@@ -133,6 +133,11 @@ public BaseUpdatePartitionSpec addField(String name, Term term) {
     Pair<Integer, String> validationKey = Pair.of(sourceTransform.first(), sourceTransform.second().toString());
 
     PartitionField existing = transformToField.get(validationKey);
+    if (existing != null && deletes.contains(existing.fieldId()) &&
+        existing.transform().toString().equals(sourceTransform.second().toString())) {
+      deletes.remove(existing.fieldId());
+      return this;
+    }
```

Comment:
Nit: missing empty line between a control flow block and the following statement.

Reply:
updated